trin-metrics/src/utils.rs

```diff
@@ -0,0 +1,135 @@
+use std::time::Instant;
+
+use prometheus_exporter::prometheus::Histogram;
+
+/// Timer to measure and record the duration of an event.
+///
+/// This timer can be stopped and observed at most once manually.
+/// Alternatively, if it isn't manually stopped it will be discarded in order to not record its
+/// value.
+#[must_use = "Timer should be kept in a variable otherwise it cannot observe duration"]
+#[derive(Debug)]
+pub struct CustomHistogramTimer {
```

Comment:
This name isn't great imo. The `Custom` prefix doesn't really give any information about the actual type. aka. All it really communicates is that this is a "custom" implementation. I think a better name would answer why are we using a "custom" implementation. But... this is a tricky one. I'm not coming up with any better names tbh. `DropableHistogramTimer` (well `dropable` isn't a word, and the regular histogram timer is also "dropable") / `UnobservedDropHistogramTimer` / `HistogramTimerWithDiscardedDrop` / `HistogramTimerWithUnobservedDrops`....

Idk, the last one is probably my favorite, but I'll leave this one up to you, at the very least the docstring above is very useful for understanding the purpose of the custom timer. Ideally, the name would also communicate the purpose of the custom timer, but there is a tradeoff here against the wordiness of the name.

Reply:
suggestion: `DiscardOnDropHistogramTimer`

Reply:
Dropable is totally a word. If it isn't how did you say it xD